### PR TITLE
feat: add components-angular as output-target from components package

### PR DESCRIPTION
**Pull request by oliverschuerch**
Renamed the previous `components-angular` package, which was used as angular workspace for the `@swisspost/design-system-intranet-header` package to `intranet-header-workspace`.

- [x] depending on #2072
- [x] depending on #2214
- [x] create tickets to add components tests (#2215)

### CODE REVIEW

File: packages/components-angular/projects/components/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// Other imports ....
+import { ComponentsModule } from '@swisspost/design-system-components-angular';
+
+@NgModule({
+  imports: [ComponentsModule],
+})
+export class AppModule {}
+```
+
+In your templates, the componets are available as:
```
Comment (imagoiq) [inline packages/components-angular/projects/components/README.md:31]:
```suggestion
In your templates, the components are available as:
```

File: packages/components-angular/projects/consumer-app/src/app/app.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const app = fixture.componentInstance;
+    expect(app.title).toEqual('consumer-app');
+  });
+
+  it('should render title', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    fixture.detectChanges();
+    const compiled = fixture.nativeElement as HTMLElement;
+    expect(compiled.querySelector('.content span')?.textContent).toContain('consumer-app app is running!');
+  });
```
Comment (imagoiq) [inline packages/components-angular/projects/consumer-app/src/app/app.component.spec.ts:28]: Maybe add more basic tests on the components to be sure that they are well rendered

File: packages/documentation/.storybook/helpers/register-web-components.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import postComponentsDocJson from '@swisspost/design-system-components/dist/docs
 import internetHeaderDocJson from '@swisspost/internet-header/dist/docs.json';
 
 defineInternetHeader(window);
-definePostComponent(window);
+definePostComponents(window);
```
Comment (imagoiq) [inline packages/documentation/.storybook/helpers/register-web-components.ts:12]: Perhaps this change should be announced in the changelog.

File: packages/documentation/src/stories/getting-started/components-angular/angular-app-component.sample.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<!-- app.component.html -->
+
+<post-icon *ngIf="condition" [attr.name]="iconName"></post-icon>
```
Comment (imagoiq) [inline packages/documentation/src/stories/getting-started/components-angular/angular-app-component.sample.html:3]: Is `attr.name` correct? I thought that it will be converted to an Angular prop.

File: packages/documentation/src/stories/getting-started/components/components.docs.mdx [hunk 5]
```diff
@@ -12,26 +13,35 @@ import SampleJsSpecific from './components-js-specific.sample?raw';
 
 [![npm version](https://badge.fury.io/js/@swisspost%2Fdesign-system-components.svg)](https://badge.fury.io/js/@swisspost%2Fdesign-system-components)
 
-A set of standard web components for easy integration with every framework or no framework at all.
+A set of standard web-components for easy integration with every framework or no framework at all.<br/>
```
Comment (imagoiq) [inline packages/documentation/src/stories/getting-started/components/components.docs.mdx:16]:
```suggestion
A set of standard Web Components for easy integration with every framework or no framework at all.<br/>
```
It's correct without dash: https://developer.mozilla.org/en-US/docs/Web/API/Web_Components. Maybe we can capitalize the words instead.

**Review by imagoiq — APPROVED** [review]

File: .changeset/stupid-keys-brush.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'@swisspost/design-system-documentation': minor
+---
+
+Added getting-started docs page for components-angular package. Updated the getting-started page of the components package.
```
[the remark this answers is not in the document]
Comment (gfellerph) [reply .changeset/stupid-keys-brush.md:5]: The info that a page was added is useful here. As for the second line, @oliverschuerch, please specify what was changed with the update so it's clear from the changelog and people don't have to go search and diff the docs. This could also be a separate changeset.

File: packages/components-angular/.eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{
+  "root": true,
+  "ignorePatterns": ["projects/**/*"],
+  "overrides": [
+    {
+      "files": ["*.ts"],
+      "parserOptions": {
+        "project": ["tsconfig.json"],
+        "createDefaultProgram": true
```
Comment (gfellerph) [inline packages/components-angular/.eslintrc.json:9]: Not sure if this will work for IntelliJ and vscode where we also had to specify the base directory dynamically with `__dirname`. Did you check @imagoiq?

File: packages/components-angular/LICENSE [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (gfellerph) [inline packages/components-angular/LICENSE:190]:
```suggestion
   Copyright 2023 Swiss Post, Ltd.
```

File: packages/components-angular/package.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "design-system-components-angular-workspace",
```
Comment (gfellerph) [inline packages/components-angular/package.json:2]: Always namespace package names for security reasons (dependency confusion). Even if they are not published.

```suggestion
  "name": "@swisspost/design-system-components-angular-workspace",
```

File: packages/components-angular/package.json [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    "karma-coverage": "~2.2.0",
+    "karma-jasmine": "~5.1.0",
+    "karma-jasmine-html-reporter": "~2.1.0",
+    "ng-packagr": "16.2.0",
+    "rimraf": "5.0.1",
+    "typescript": "~5.1.3"
+  },
+  "peerDependencies": {
+    "@swisspost/design-system-components": "workspace:1.5.1"
+  }
```
Comment (gfellerph) [inline packages/components-angular/package.json:48]: Why is this a peer dependency?

File: packages/components-angular/projects/components/CONTRIBUTING.md
Comment (gfellerph) [inline packages/components-angular/projects/components/CONTRIBUTING.md:1]: Please mention that this package is auto generated by stencil and to make changes people need to go to the components package.

File: packages/components-angular/projects/components/package.json [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  "homepage": "https://swisspost-web-frontend.netlify.app",
+  "bugs": {
+    "url": "https://github.com/swisspost/design-system/issues"
+  },
+  "private": true,
+  "files": [
+    "dist"
+  ],
+  "publishConfig": {
+    "access": "restricted",
```
Comment (gfellerph) [inline packages/components-angular/projects/components/package.json:20]: Is the private flag here intentional? With `access: restricted` and `private: true`, the components don't get published.

File: packages/intranet-header-workspace/CHANGELOG.md
Comment (gfellerph) [inline packages/intranet-header-workspace/CHANGELOG.md:1]: Why was this deleted?

**Review by gfellerph — CHANGES_REQUESTED** [review]
Thank you so much for pushing this. This is a cornerstone for the Design System.

File: .changeset/pretty-ligers-bathe.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+---
+'@swisspost/design-system-components-angular': minor
```
Comment (gfellerph) [inline .changeset/pretty-ligers-bathe.md:2]: This will release as 1.0.0 (see change request in package.json), which gives a slightly better changelog, starting with v1.0.0 instead of v1.1.0.

```suggestion
'@swisspost/design-system-components-angular': major
```

File: packages/components-angular/projects/components/package.json [hunk 13]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@swisspost/design-system-components-angular",
+  "version": "1.0.0",
```
Comment (gfellerph) [inline packages/components-angular/projects/components/package.json:3]:
```suggestion
  "version": "0.0.1",
```

File: packages/documentation/src/stories/getting-started/components/components.docs.mdx
Comment (gfellerph) [inline packages/documentation/src/stories/getting-started/components/components.docs.mdx:1]: There might have been a misunderstanding here. I tried to split between users that develop with a framework & bundler and users that need CDN integration since these approaches are very different.

I expect that 95% of users need the framework integration docs, so they should come first. At the moment, the CDN docs are first and might be confusing for the majority of people.

File: .changeset/pretty-ligers-bathe.md [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+---
+
+We are introducing the new package `@swisspost/design-system-components-angular` 🥳, which provides a corresponding Angular component for all our components.
```
Comment (gfellerph) [inline .changeset/pretty-ligers-bathe.md:5]:
```suggestion
We are introducing the new package `@swisspost/design-system-components-angular` 🥳, which provides a corresponding Angular component for all our web-components.
```

**Review by gfellerph — CHANGES_REQUESTED** [review]
Generally looking good, but the getting started docs need some improvement in my eyes.

File: packages/components-angular/.eslintrc.json (hunk 7 above)
Comment (imagoiq) [reply packages/components-angular/.eslintrc.json:9]: It looks ok at first sight at least. I didn't see any errors.

**Review by gfellerph — APPROVED** [review]

**Review by alizedebray — APPROVED** [review]

**Comment by oliverschuerch** [comment]
PR to fix the unit tests is on the way: #2424

File: packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import AngularAppModuleSample from './angular-app-module.sample.ts?raw';
+import AngularAppComponentSample from './angular-app-component.sample.html?raw';
+
+<Meta of={GettingStartedStories} />
+
+# @swisspost/design-system-components-angular
+
+[![npm version](https://badge.fury.io/js/@swisspost%2Fdesign-system-components-angular.svg)](https://badge.fury.io/js/@swisspost%2Fdesign-system-components-angular)
+
+A set of generated wrapper components around our web-components, for easy integration with the Angular framework.
```
Comment (gfellerph) [inline packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx:12]: Not sure if this is a selling point. I'd just talk about Angular components.
```suggestion
The Design System Angular Components, ready for use in your Angular project.
```

File: packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+## Installation
+
+<Source code={`npm install @swisspost/design-system-components-angular`} language="bash" />
+
+## Usage
+
+Now you can reference our angular components as a standard import.
```
Comment (gfellerph) [inline packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx:20]:
```suggestion
Importing the components to your project.
```

File: packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+
+<Source code={AngularAppModuleSample} language="typescript" />
+
+You can now directly leverage our components in their template and take advantage of Angular template binding syntax.
```
Comment (gfellerph) [inline packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx:24]: Usage within your template.

**Review by gfellerph — APPROVED** [review]
Mainly suggestions, therefore I'll approve and leave it up to you to implement.

File: packages/components/src/components/post-popovercontainer/post-popovercontainer.scss [hunk 18]
```diff
@@ -11,7 +11,11 @@
 // Can be removed as soon as popover is supported by all major browsers
 // https://caniuse.com/?search=popover
 @layer polyfill {
-  @include meta.load-css('@oddbird/popover-polyfill/dist/popover.css');
+  @supports not selector(:popover-open) {
+    [popover]:not(.\:popover-open) {
+      display: none;
+    }
+  }
```
Comment (imagoiq) [inline packages/components/src/components/post-popovercontainer/post-popovercontainer.scss:18]: Why removing this polyfill ? Why in this PR ? Maybe this is wrong as the package is still installed ?

File: packages/documentation/src/stories/getting-started/components-angular/angular-app-component.sample.html (hunk 4 above)
Comment (imagoiq) [reply packages/documentation/src/stories/getting-started/components-angular/angular-app-component.sample.html:3]: In my testing both seems to work. Is there a preferred way? Do we want to document it somewhere?

File: packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx (hunk 15 above)
Comment (imagoiq) [reply packages/documentation/src/stories/getting-started/components-angular/angular.docs.mdx:12]: In my opinion, it's worth it to be honest and keep mentioning that there are only wrappers around web-components and not Angular components made manually.
